community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseBuilder.java

```diff
@@ -118,6 +119,16 @@ public GraphDatabaseBuilder setConfig( Map<String, String> config )
     }
 
     /**
+     * Retrieve the configured configuration values.
+     *
+     * @return the configured configuration values.
+     */
+    public Config config()
+    {
+        return new Config( config );
+    }
```

Comment:
@thobe I am not happy about adding this to the public API. The Config object without the settings classes is usable but doesn't provide all the functionality you would expect of that object (e.g. validation, migration). Also `Config` itself isn't part of the public API.